pandas/_libs/tslibs/np_datetime.pyx

```diff
@@ -273,14 +274,25 @@ cdef inline int string_to_dts(
     int* out_local,
     int* out_tzoffset,
     bint want_exc,
+    format: str | None = None,
+    bint exact = True,
 ) except? -1:
     cdef:
         Py_ssize_t length
         const char* buf
+        Py_ssize_t format_length
+        const char* format_buf
 
     buf = get_c_string_buf_and_size(val, &length)
+    if format is None:
+        format_buf = b''
+        format_length = 0
+        exact = False
```

Comment:
do we need to set `exact` to `False` if format is empty but `exact=True` is provided?

Reply:
If we stick with the `to_datetime` defaults of `format=None` and `exact=True`, then in the case of `format=None` there's no format to respect, and having an empty `format_buf` with `exact=True` would fail on any non-empty date-string

I think `format=None` should be interpreted as "don't match `format`", in which case if the C function doesn't accept `None` and we need to pass an empty string, then we'd need to set `exact` to `False`